rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/rest/TbHttpClient.java

```diff
@@ -63,7 +69,25 @@
             if (config.getMaxParallelRequestsCount() > 0) {
                 pendingFutures = new ConcurrentLinkedDeque<>();
             }
-            if (config.isUseSimpleClientHttpFactory()) {
+
+            if (config.isEnableProxy()) {
+                SimpleClientHttpRequestFactory factory = new SimpleClientHttpRequestFactory();
```

Comment:
add validation of config and meaningful exception

Reply:
done